samples/streaming/consumer/src/main.rs

```diff
@@ -20,32 +20,35 @@ use std::error::Error;
 use std::io::Cursor;
 use tokio_stream::StreamExt;
 use tonic::transport::Channel;
-use uuencode::uudecode;
 
-async fn streaming(
+/// Stream images from the server and display them in the provided window.
+///
+/// # Arguments
+/// * `client` - The client connection to the service that will transfer the stream.
+/// * `number_of_images` - The number of images that we will stream.
+/// * `window` - The window where the streamed images will be shown.
+async fn stream_images(
     client: &mut DigitalTwinProviderClient<Channel>,
     entity_id: &str,
-    num: usize,
+    number_of_images: usize,
     window: &mut WindowProxy,
 ) -> Result<(), Box<dyn Error>> {
     let stream =
         client.stream(StreamRequest { entity_id: entity_id.to_string() }).await?.into_inner();
 
-    // The stream is infinite, so take just num elements and then disconnect.
-    let mut stream = stream.take(num);
+    // The stream is infinite, so we will just take number_of_images elements and then disconnect.
+    let mut stream = stream.take(number_of_images);
     while let Some(item) = stream.next().await {
-        let content = item.unwrap().content;
-        if let Some((contents, filename)) = uudecode(&content) {
-            let image_reader = ImageReader::new(Cursor::new(contents)).with_guessed_format()?;
-            let image = image_reader.decode()?;
-            let image_data = image.as_bytes().to_vec();
-            let image_view =
-                ImageView::new(ImageInfo::rgb8(image.width(), image.height()), &image_data);
-            window.set_image(filename, image_view)?;
-        }
+        let media_content = item.unwrap().media.unwrap().media_content;
```

Comment:
can `media` be `None`? you might want to check before unwrapping

Reply:
Fixed.